python/ray/llm/_internal/batch/observability/usage_telemetry/usage.py

```diff
@@ -0,0 +1,101 @@
+from enum import Enum
+from typing import Callable, Dict, Optional
+
+import ray
+
+from ray._private.usage.usage_lib import record_extra_usage_tag
+
+from ray.llm._internal.serve.observability.usage_telemetry.usage import (
+    RAYLLM_TELEMETRY_NAMESPACE,
+)
+from ray.llm._internal.batch.observability.logging import get_logger
+
+RAYLLM_BATCH_TELEMETRY_ACTOR_NAME = "rayllm_batch_telemetry"
+
+logger = get_logger(__name__)
+
+
+class BatchTelemetryTags(str, Enum):
+    """Telemetry tags for RayLLM Batch."""
+
+    LLM_BATCH_PROCESSOR_CONFIG_NAME = "LLM_BATCH_PROCESSOR_CONFIG_NAME"
+    LLM_BATCH_MODEL_SOURCE = "LLM_BATCH_MODEL_SOURCE"
+    LLM_BATCH_SIZE = "LLM_BATCH_SIZE"
+    LLM_BATCH_ACCELERATOR_TYPE = "LLM_BATCH_ACCELERATOR_TYPE"
+    LLM_BATCH_CONCURRENCY = "LLM_BATCH_CONCURRENCY"
+    LLM_BATCH_TASK_TYPE = "LLM_BATCH_TASK_TYPE"
+    LLM_BATCH_PIPELINE_PARALLEL_SIZE = "LLM_BATCH_PIPELINE_PARALLEL_SIZE"
+    LLM_BATCH_TENSOR_PARALLEL_SIZE = "LLM_BATCH_TENSOR_PARALLEL_SIZE"
+
+
+@ray.remote(
+    name=RAYLLM_BATCH_TELEMETRY_ACTOR_NAME,
+    namespace=RAYLLM_TELEMETRY_NAMESPACE,
+    num_cpus=0,
+    lifetime="detached",
+)
+class _TelemetryAgent:
+    """Named Actor to keep the state of all deployed models and record telemetry."""
+
+    def __init__(self):
+        self.record_tag_func = record_extra_usage_tag
+
+    def update_record_tag_func(self, record_tag_func: Callable) -> None:
```

Comment:
If this is only meant to be used internally, let's add a leading underscore in front of the function name
```suggestion
    def _update_record_tag_func(self, record_tag_func: Callable) -> None:
```